### ISSUE TITLE
HTTP/2 and/or HTTP/3?

### ISSUE THREAD

**Issue by jackfirth** [Bogdanp/racket-http-easy#1]
Is support for HTTP/2 and/or HTTP/3 on the roadmap for this package? If it is, would you like help with that?

**Comment by Bogdanp** [comment]
Yes, I think it'd be a good idea to support those. I was thinking it would be great if we came up with a standard interface for these protocols and then have separate packages, like h11 and h2 in Python, that just implement the protocol parsing bits, then any library could take them and handle the higher-level stuff like content negotiation, upgrading, etc.

A note on the design of these protocol libs: in Python, these libraries are forced to operate on bytes rather than sockets due to the multiple IO systems. I think in Racket we'd just have them operate directly on ports.

My goal right now is to get the library to a point where I no longer have to write [code like this](https://github.com/Bogdanp/racket-sentry/blob/9794b2da9c4f3ca8c8094d6bc78d5ca8bf9b133b/sentry-lib/sentry.rkt#L102-L147) myself so I don't plan to work on any of the protocol stuff for the time being. Feel free to take the lead on it!
